pkg/reconciler/taskrun/resources/taskspec.go

```diff
@@ -32,6 +32,10 @@ type ResolvedTask struct {
 	TaskName string
 	Kind     v1beta1.TaskKind
 	TaskSpec *v1beta1.TaskSpec
+	// PipelineResolvedTask is used to store the Pipeline resolved task for trusted resources verification
+	PipelineResolvedTask *v1beta1.Task
+	// RefSource is used to store the RefSource of Pipeline task
+	RefSource *v1beta1.RefSource
```

Comment:
Same question as before - how should callers interpret the difference between overlapping fields `TaskSpec` and `PipelineResolvedTask.TaskSpec`, `TaskName` and `PipelineResolvedTask.Name`, etc.?